### PR TITLE
Randomize sync times

### PR DESCRIPTION
**Pull request by dpsutton**
Randomizes metabase managed sync schedules.

there are two schedules at play: the hourly sync task and the daily field values task.

- when manged by metabase, choose a random minute for the hourly and a random hour for the field values task
- when starting up, look to see if the schedules are manged by metabase, and if the old default (hourly at 50 or 0 (the db had a default of hourly at 50 and the application had a default of hourly at 0)) and if so, randomize them. This was done to not need to figure out how to insert random values during a migration but just do it on application startup. This also allows a spot for more advanced scheduling techniques in the future: ie, take into account how much time each takes and try to schedule them with respect to this.
- put a minute picker on the UI for scheduling

### CODE REVIEW

**Review by sbelak — APPROVED** [review]
Seems reasonable and a straight improvement of status quo. 

At some point it would be dope if we had probabilistic sync based on underlying change frequency (but this is more a note to self).

File: src/metabase/api/database.clj [hunk 1]
```diff
@@ -572,18 +551,30 @@
...
+                                                      :is_full_sync       full-sync?
+                                                      :is_on_demand       (boolean is_on_demand)
+                                                      :description        description
+                                                      :caveats            caveats
+                                                      :points_of_interest points_of_interest
+                                                      :auto_run_queries   auto_run_queries}
+                                                     (cond
+                                                       ;; transition back to metabase managed schedules. the schedule
+                                                       ;; details, even if provided, are ignored
+                                                       (and (get-in database [:details :let-user-control-scheduling])
```
[the remark this answers is not in the document]
Comment (dpsutton) [reply src/metabase/api/database.clj:567]: details are the incoming new details. the `database` is the old one currently in the database. It's looking to see if its going back to metabase managed schedules by detecting an edge change of `:let-user-control-schedules` going from true to false.

[the remark this answers is not in the document]
Comment (sbelak) [reply src/metabase/api/database.clj:567]: Makes sense. Perhaps name `details` in a way that indicates that so we don't get drive-by changes by people like me who missed that.

[the remark this answers is not in the document]
Comment (dpsutton) [reply src/metabase/api/database.clj:568]: i added some more comment. those are destructured bindings so don't have a ton of leeway without introducing a bunch of changes `new-details :details` etc. oh, maybe renaming database to existing-database or something similar can make the point with a smaller footprint of change. good thinking @sbelak

**Pull request merged**